crawl4ai/utils.py

```
@@ -1991,24 +1991,101 @@ def fast_format_html(html_string):
+
+        return cleaned_url
+    except (ValueError, AttributeError, TypeError) as e:
+        print(f"Error cleaning URL '{_url}': {e}")
+        try:
+            # More robust fallback that preserves the original URL structure
+            _ss = _url.split('//')
+            if len(_ss) == 2:
+                return '//'.join(_ss)
+            else:
+                return _ss[0] + '//' + '/'.join(_ss[1:])
+        except:
```

Ruff (0.8.2)

Do not use bare `except`

(E722)